src/editor/tests.py

```diff
@@ -756,39 +756,52 @@ def test_catalog(self):
         except:
             found = False
         self.assertEqual(found, True)
-        resp = self.client.post(reverse('add_contributor',
-                                        kwargs={'submission_id': self.book.id,
-                                                'contributor_type': 'author'}),
-                                {'salutation': 'Mr',
-                                 'first_name': 'rua_first_new_name',
-                                 'last_name': 'rua_last_new_name',
-                                 'middle_name': '', 'biography': 'updated bio',
-                                 'institution': 'rua_testing', 'department': '',
-                                 'country': 'GB', 'orcid': '', 'twitter': '',
-                                 'facebook': '', 'linkedin': '',
-                                 'author_email': 'fake_changed@fakeaddress.com'})
-        found = False
+        resp = self.client.post(
+            reverse('add_contributor',
+                    kwargs={'submission_id': self.book.id,
+                            'contributor_type': 'author'}),
+            {'salutation': 'Mr',
+             'first_name': 'rua_first_new_name',
+             'last_name': 'rua_last_new_name',
+             'middle_name': '',
+             'biography': 'updated bio',
+             'institution': 'rua_testing',
+             'department': '',
+             'country': 'GB',
+             'orcid': '', 'twitter': '',
+             'facebook': '', 'linkedin': '',
+             'author_email': 'fake_changed@fakeaddress.com'}
+        )
+
+        self.assertEqual(resp.status_code, 302)
+        found = True
         try:
-            author = core_models.Author.objects.get(pk=2)
-            found = True
+            # Get new author without relying on hard coded pk
+            author = core_models.Author.objects.get(
+                first_name='rua_first_new_name'
+            )
         except:
             found = False
         self.assertEqual(found, True)
         self.assertEqual(author.first_name, 'rua_first_new_name')
         self.assertEqual(author.last_name, 'rua_last_new_name')
-        resp = self.client.post(reverse('delete_contributor',
-                                        kwargs={'submission_id': self.book.id,
-                                                'contributor_type': 'author',
-                                                'contributor_id': author.pk}))
+        resp = self.client.post(
+            reverse('delete_contributor',
+            kwargs={'submission_id': self.book.id,
+                    'contributor_type': 'author',
+                    'contributor_id': author.pk})
+        )
         found = False
```

Comment:
It might be worth redoing the change you made above: Setting `found  = True` before the `try/except` block of code.
This avoids the `found  = False` in the except statement being redundant (if that makes sense?)

Reply:
Well spotted :-)